libs/datamodel/core/src/configuration/datasource.rs

```diff
@@ -40,6 +43,66 @@ impl Datasource {
         let capabilities = self.active_connector.capabilities().clone();
         ConnectorCapabilities::new(capabilities)
     }
+
+    /// Sigh. I'm sorry.
+    ///
+    /// JavaScript wants to set the CWD, but can't due to changing it for itself
+    /// too. This method is an ugly hack to convert all relative paths in the
+    /// connection string to include a given config directory as its prefix, but
+    /// only if the path is relative.
+    ///
+    /// The other option would've been to just point in our docs that a path
+    /// in a connection string is relative only from the caller point of view.
+    ///
+    /// Now, there is a promise to change this behavior to follow how everybody
+    /// else is doing it, but I wouldn't hold my breath.
+    ///
+    /// Written by Julius de Bruijn in the holy date of 25.3.2021 for the future
+    /// generations.
+    ///
+    /// P.S. Don't forget to add new parameters here if needed!
+    pub fn set_config_dir(&mut self, config_dir: &Path) {
+        let set_root = |path: &str| {
+            let path = Path::new(path);
+
+            if path.is_relative() {
+                Some(config_dir.join(&path).to_str().map(ToString::to_string).unwrap())
+            } else {
+                None
+            }
+        };
+
+        match self.active_provider.as_str() {
+            "sqlserver" => (),
+            "sqlite" => {
+                if let Some(path) = set_root(&self.url.value.trim_start_matches("file:")) {
+                    self.url.value = format!("file:{}", path);
+                };
+            }
+            _ => {
+                let mut url = Url::parse(&self.url.value).unwrap();
+
+                let mut params: BTreeMap<String, String> =
+                    url.query_pairs().map(|(k, v)| (k.to_string(), v.to_string())).collect();
+
+                url.query_pairs_mut().clear();
+
+                if let Some(path) = params.get("sslcert").map(|s| s.as_str()).and_then(set_root) {
+                    params.insert("sslcert".into(), path.into());
+                }
+
+                if let Some(path) = params.get("sslidentity").map(|s| s.as_str()).and_then(set_root) {
+                    params.insert("sslidentity".into(), path.into());
+                }
+
+                for (k, v) in params.into_iter() {
+                    url.query_pairs_mut().append_pair(&k, &v);
+                }
+
+                self.url.value = url.to_string();
+            }
+        }
+    }
```

Comment:
I think we need product to be aware of that, in particular with the coming environment management work. This is for sure going to cause bugs in the future.